### DISCUSSION TITLE
Don't understand what is required to "clone from server", github url yields errors, i'm sure it's my naivety

### DISCUSSION THREAD

**Discussion by RoryWatts** [zeapo/Android-Password-Store#2159] [Q&A] [answered]
In the first-time setup, the wiki suggests to select "Clone from server" to clone a store from a remote server, which is what I want to do.

I have my pass directory on github (literally just pushing the whole directory), but I'm not sure that this arrangement qualifies as having a server as implied in the wiki.

Regardless, I have tried to input the repository URL in the format as follows: `https://github.com/RoryWatts/my-password-repo.git`
but this brings up the error "Potentially incorrect URL", and asks me to specify the HTTPS username in the form `https://username@example.com/...`
I then tried to input the repository URL in this form as `https://RoryWatts@github.com/my-password-repo.git`, but this yields an error `https://RoryWatts@github.com/my-password-repo.git/info/refs?service=git-upload-pack not found`

My best guess is the problem is one of the two following:
- My repo being uploaded to github is not what is meant by a server, and I need to do some other step
- My github repo is a server, but I have not specified the repository URL in the format correctly.

password-store has been such an excellent tool and i'm really hoping I can use it on my android too, so please forgive my naivety about the set-up, and I hope that someone might be able to help me!

**Comment by msfjarvis** [comment, marked as answer]
When you add your username you shouldn't remove it from the repository path, so the correct URL would be `https://RoryWatts@github.com/RoryWatts/my-password-repo.git`.

I can understand why this confusion can arise, I'll see if we can somehow improve the flow around this.

**Reply by RoryWatts** [reply]
Thanks. I suppose my confusion came from not having seen this URL format beforehand.
